hw4/04-templates.js

```diff
@@ -14,51 +14,77 @@ const url = 'https://restcountries.com/v3.1/all';
 // Add your code here
 
 app.get('/', (req, res) => {
-  // render pug template for the index.html file
-
-  res.render('index', {
-    heading: 'Countries of the World',
-    main: 'Welcome to this application. Using the REST Countries API, we will be showing the countries and capitals of the world, the most populous countries in the world, and the number of countries in each region of the world',
-  });
+    // render pug template for the index.html file
+    res.render('index', {
+        heading: 'Countries of the World',
+        main: 'Welcome to this application. Using the REST Countries API, we will be showing the countries and capitals of the world, the most populous countries in the world, and the number of countries in each region of the world',
+    });
 });
 
-app.get('/capitals', (req, res) => {
-  // map the output array to create an array with country names and capitals
-  // check for empty data in the output array
-
-  let countries = ['Afghanistan', 'Aland Islands', 'Albania'];
-
-  res.render('page', {
-    heading: 'Countries and Capitals',
-    results: countries,
-  });
+app.get('/capitals', async (req, res) => {
+    // map the output array to create an array with country names and capitals
+    // check for empty data in the output array
+    getCountries((data) => {
+        let countryAndCapitals = data
+            .map((response) => {
+                return `${response.name.common ? response.name.common : 'Not found'} - ${
+                    response.capital ? response.capital : 'Not found'
+                }`;
+            })
+            .sort();
+
+        res.render('page', {
+            heading: 'Countries and Capitals',
+            results: countryAndCapitals,
+        });
+    }).catch((err) => console.log(err));
 });
 
 app.get('/populous', (req, res) => {
-  // filter the output array for the countries with population of 50 million or more
-  // sort the resulting array to show the results in order of population
-  // map the resulting array into a new array with the country name and formatted population
-
-  let populous = ['China', 'India', 'United States of America'];
-
-  res.render('page', {
-    heading: 'Most Populous Countries',
-    results: populous,
-  });
+    // filter the output array for the countries with population of 50 million or more
+    // sort the resulting array to show the results in order of population
+    // map the resulting array into a new array with the country name and formatted population
+    getCountries((data) => {
+        const result = data
+            .filter((response) => response.population > 50000000)
+            .map((response) => {
+                return `${response.name.common} - ${response.population}`;
+            })
+            .sort();
```

Comment:
This is sorting by country name not by population size. Also pls format the population number